lib/ecto/adapters/postgres.ex

```diff
@@ -124,7 +124,7 @@ defmodule Ecto.Adapters.Postgres do
   def storage_up(opts) do
     database = Keyword.fetch!(opts, :database) || raise ":database is nil in repository configuration"
     encoding = opts[:encoding] || "UTF8"
-    opts     = Keyword.put(opts, :database, "postgres")
+    opts     = Keyword.put(opts, :database, opts[:default_database] || "postgres")
```

Comment:
What do you think about naming this option as `maintenance_database`? [That's the nomenclature `createdb` uses](https://www.postgresql.org/docs/9.3/static/app-createdb.html).

```elixir
maintenance_database = Keyword.get(opts, :maintenance_database, "postgres")
opts = Keyword.put(opts, :database, maintenance_database)
```

Reply:
Yes, that makes sense. I can fix it.